### PR TITLE
fix(curriculum): standardize links in curriculum

### LINKED ISSUE #39739: Standardize links in curriculum
The curriculum (project and challenge pages) contain links to various places; github, codepen, replit, etc. Sometimes they open in a new tab, sometimes not. Sometimes they contain `rel="noopener noreferrer"`, sometimes not.

We could go through and make them all consistent? Adding `target="_blank" rel="noopener noreferrer nofollow"` to all of them?

[Here's a good article explaining the `rel` values](https://pointjupiter.com/what-noopener-noreferrer-nofollow-explained/)

Not sure what we all want to use. I think we could make them all open in a new tab at the least.

----

Edit: 

For links with these format, `[hexadecimal numbers here](https://www.freecodecamp.org/news/hexadecimal-number-system/)` regex `/\[(.*?)\]\((.*?)\)/gm` should target them and `<a href="$2"target="_blank" rel="noopener noreferrer nofollow">$1</a>` will swap them into `<a href="https://www.freecodecamp.org/news/hexadecimal-number-system/"target="_blank" rel="noopener noreferrer nofollow">hexadecimal numbers here</a>`

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #39739

This should be all of them, if I missed something or made a mistake, please say so

### CODE REVIEW

File: curriculum/challenges/english/01-responsive-web-design/basic-css/use-hex-code-for-specific-colors.md [hunk 1]
```diff
@@ -11,9 +11,9 @@ dashedName: use-hex-code-for-specific-colors
...
 
-We usually use <dfn>decimals</dfn>, or base 10 numbers, which use the symbols 0 to 9 for each digit. <dfn>Hexadecimals</dfn> (or <dfn>hex</dfn>) are base 16 numbers. This means it uses sixteen distinct symbols. Like decimals, the symbols 0-9 represent the values zero to nine. Then A,B,C,D,E,F represent the values ten to fifteen. Altogether, 0 to F can represent a digit in hexadecimal, giving us 16 total possible values. You can find more information about [hexadecimal numbers here](https://www.freecodecamp.org/news/hexadecimal-number-system/).
+We usually use <dfn>decimals</dfn>, or base 10 numbers, which use the symbols 0 to 9 for each digit. <dfn>Hexadecimals</dfn> (or <dfn>hex</dfn>) are base 16 numbers. This means it uses sixteen distinct symbols. Like decimals, the symbols 0-9 represent the values zero to nine. Then A,B,C,D,E,F represent the values ten to fifteen. Altogether, 0 to F can represent a digit in hexadecimal, giving us 16 total possible values. You can find more information about <a href="https://www.freecodecamp.org/news/hexadecimal-number-system/"target="_blank" rel="noopener noreferrer nofollow">hexadecimal numbers here</a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/01-responsive-web-design/basic-css/use-hex-code-for-specific-colors.md:14]: Missed a space _everywhere_

```suggestion
We usually use <dfn>decimals</dfn>, or base 10 numbers, which use the symbols 0 to 9 for each digit. <dfn>Hexadecimals</dfn> (or <dfn>hex</dfn>) are base 16 numbers. This means it uses sixteen distinct symbols. Like decimals, the symbols 0-9 represent the values zero to nine. Then A,B,C,D,E,F represent the values ten to fifteen. Altogether, 0 to F can represent a digit in hexadecimal, giving us 16 total possible values. You can find more information about <a href="https://www.freecodecamp.org/news/hexadecimal-number-system/" target="_blank" rel="noopener noreferrer nofollow">hexadecimal numbers here</a>.
```

Comment (Sboonny) [reply curriculum/challenges/english/01-responsive-web-design/basic-css/use-hex-code-for-specific-colors.md:14]: On it👍

**Comment by Sboonny** [comment]
Sure, if I face any issues, I will discuss it in discord

**Review by moT01 — DISMISSED** [review]
LGTM 🎉

File: curriculum/challenges/english/01-responsive-web-design/css-flexbox/align-elements-using-the-justify-content-property.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ dashedName: align-elements-using-the-justify-content-property
 
 Sometimes the flex items within a flex container do not fill all the space in the container. It is common to want to tell CSS how to align and space out the flex items a certain way. Fortunately, the `justify-content` property has several options to do this. But first, there is some important terminology to understand before reviewing those options.
 
-[For more info about flex-box properties](https://www.freecodecamp.org/news/flexbox-the-ultimate-css-flex-cheatsheet/)
+<a href="https://www.freecodecamp.org/news/flexbox-the-ultimate-css-flex-cheatsheet/" target="_blank" rel="noopener noreferrer nofollow">For more info about flex-box properties</a>
```
Comment (ieahleen) [inline curriculum/challenges/english/01-responsive-web-design/css-flexbox/align-elements-using-the-justify-content-property.md:14]: The sentence looks uncomplete

For more info about flex-box properties read here?
```suggestion
<a href="https://www.freecodecamp.org/news/flexbox-the-ultimate-css-flex-cheatsheet/" target="_blank" rel="noopener noreferrer nofollow">For more info about flex-box properties</a> read here.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ dashedName: assignment-with-a-returned-value
 
 # --description--
 
-If you'll recall from our discussion of [Storing Values with the Assignment Operator](/learn/javascript-algorithms-and-data-structures/basic-javascript/storing-values-with-the-assignment-operator), everything to the right of the equal sign is resolved before the value is assigned. This means we can take the return value of a function and assign it to a variable.
+If you'll recall from our discussion of <a href="/learn/javascript-algorithms-and-data-structures/basic-javascript/storing-values-with-the-assignment-operator" target="_blank" rel="noopener noreferrer nofollow">Storing Values with the Assignment Operator</a>, everything to the right of the equal sign is resolved before the value is assigned. This means we can take the return value of a function and assign it to a variable.
```
Comment (ieahleen) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/assignment-with-a-returned-value.md:12]:
```suggestion
If you'll recall from our discussion about <a href="/learn/javascript-algorithms-and-data-structures/basic-javascript/storing-values-with-the-assignment-operator" target="_blank" rel="noopener noreferrer nofollow">Storing Values with the Assignment Operator</a>, everything to the right of the equal sign is resolved before the value is assigned. This means we can take the return value of a function and assign it to a variable.
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ dashedName: compare-scopes-of-the-var-and-let-keywords
 
 # --description--
 
-If you are unfamiliar with `let`, check out [this challenge](/learn/javascript-algorithms-and-data-structures/basic-javascript/explore-differences-between-the-var-and-let-keywords).
+If you are unfamiliar with `let`, check out <a href="/learn/javascript-algorithms-and-data-structures/basic-javascript/explore-differences-between-the-var-and-let-keywords" target="_blank" rel="noopener noreferrer nofollow">this challenge</a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/compare-scopes-of-the-var-and-let-keywords.md:11]:
```suggestion
If you are unfamiliar with `let`, check out <a href="/learn/javascript-algorithms-and-data-structures/basic-javascript/explore-differences-between-the-var-and-let-keywords" target="_blank" rel="noopener noreferrer nofollow">this challenge about the difference bewteen `let` and `var`</a>.
```

more accessible link text

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/mutate-an-array-declared-with-const.md [hunk 5]
```diff
@@ -8,7 +8,7 @@ dashedName: mutate-an-array-declared-with-const
 
 # --description--
 
-If you are unfamiliar with `const`, check out [this challenge](/learn/javascript-algorithms-and-data-structures/basic-javascript/declare-a-read-only-variable-with-the-const-keyword).
+If you are unfamiliar with `const`, check out <a href="/learn/javascript-algorithms-and-data-structures/basic-javascript/declare-a-read-only-variable-with-the-const-keyword" target="_blank" rel="noopener noreferrer nofollow">this challenge</a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/mutate-an-array-declared-with-const.md:11]:
```suggestion
If you are unfamiliar with `const`, check out <a href="/learn/javascript-algorithms-and-data-structures/basic-javascript/declare-a-read-only-variable-with-the-const-keyword" target="_blank" rel="noopener noreferrer nofollow">this challenge about the `const` keyword</a>.
```

File: curriculum/challenges/english/03-front-end-development-libraries/jquery/change-text-inside-an-element-using-jquery.md [hunk 6]
```diff
@@ -22,7 +22,7 @@ jQuery also has a similar function called `.text()` that only alters text withou
 
 Change the button with id `target4` by emphasizing its text.
 
-[View our news article for &lt;em>](https://www.freecodecamp.org/news/html-elements-explained-what-are-html-tags/#em-element) to learn the difference between `<i>` and `<em>` and their uses.
+<a href="https://www.freecodecamp.org/news/html-elements-explained-what-are-html-tags/#em-element" target="_blank" rel="noopener noreferrer nofollow">View our news article for &lt;em></a> to learn the difference between `<i>` and `<em>` and their uses.
```
Comment (ieahleen) [inline curriculum/challenges/english/03-front-end-development-libraries/jquery/change-text-inside-an-element-using-jquery.md:25]:
```suggestion
<a href="https://www.freecodecamp.org/news/html-elements-explained-what-are-html-tags/#em-element" target="_blank" rel="noopener noreferrer nofollow">View our news article for &lt;em&gt;</a> to learn the difference between `<i>` and `<em>` and their uses.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/handle-a-disconnect.md [hunk 7]
```diff
@@ -22,7 +22,7 @@ To make sure clients continuously have the updated count of current users, you s
 
 **Note:** Just like `'disconnect'`, all other events that a socket can emit to the server should be handled within the connecting listener where we have 'socket' defined.
 
-Submit your page when you think you've got it right. If you're running into errors, you can check out the project completed up to this point <a href="https://gist.github.com/camperbot/ab1007b76069884fb45b215d3c4496fa" target="_blank" rel="noopener noreferrer nofollow">https://gist.github.com/camperbot/ab1007b76069884fb45b215d3c4496fa</a>.
+Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/ab1007b76069884fb45b215d3c4496fa" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point </a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/handle-a-disconnect.md:25]:
```suggestion
Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/ab1007b76069884fb45b215d3c4496fa" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point</a>.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-put-a-profile-together.md [hunk 8]
```diff
@@ -24,7 +24,7 @@ Also, in `profile.pug`, add a link referring to the `/logout` route, which will
 a(href='/logout') Logout
 ```
 
-Submit your page when you think you've got it right. If you're running into errors, you can check out the project completed up to this point <a href="https://gist.github.com/camperbot/136b3ad611cc80b41cab6f74bb460f6a" target="_blank" rel="noopener noreferrer nofollow">https://gist.github.com/camperbot/136b3ad611cc80b41cab6f74bb460f6a</a>.
+Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/136b3ad611cc80b41cab6f74bb460f6a" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point </a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/how-to-put-a-profile-together.md:27]:
```suggestion
Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/136b3ad611cc80b41cab6f74bb460f6a" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point</a>.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implement-the-serialization-of-a-passport-user.md [hunk 9]
```diff
@@ -40,7 +40,7 @@ myDB(async client => {
 
 Be sure to uncomment the `myDataBase` code in `deserializeUser`, and edit your `done(null, null)` to include the `doc`.
 
-Submit your page when you think you've got it right. If you're running into errors, you can check out the project completed up to this point <a href="https://gist.github.com/camperbot/175f2f585a2d8034044c7e8857d5add7" target="_blank" rel="noopener noreferrer nofollow">https://gist.github.com/camperbot/175f2f585a2d8034044c7e8857d5add7</a>.
+Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/175f2f585a2d8034044c7e8857d5add7" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point </a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/implement-the-serialization-of-a-passport-user.md:43]:
```suggestion
Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/175f2f585a2d8034044c7e8857d5add7" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point</a>.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/registration-of-new-users.md [hunk 10]
```diff
@@ -47,7 +47,7 @@ app.route('/register')
   );
 ```
 
-Submit your page when you think you've got it right. If you're running into errors, you can check out the project completed up to this point <a href="https://gist.github.com/camperbot/b230a5b3bbc89b1fa0ce32a2aa7b083e" target="_blank" rel="noopener noreferrer nofollow">https://gist.github.com/camperbot/b230a5b3bbc89b1fa0ce32a2aa7b083e</a>.
+Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/b230a5b3bbc89b1fa0ce32a2aa7b083e" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point </a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/registration-of-new-users.md:50]:
```suggestion
Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/b230a5b3bbc89b1fa0ce32a2aa7b083e" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point</a>.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/serialization-of-a-user-object.md [hunk 11]
```diff
@@ -28,7 +28,7 @@ passport.deserializeUser((id, done) => {
 
 NOTE: This `deserializeUser` will throw an error until we set up the DB in the next step, so for now comment out the whole block and just call `done(null, null)` in the function `deserializeUser`.
 
-Submit your page when you think you've got it right. If you're running into errors, you can check out the project completed up to this point <a href="https://gist.github.com/camperbot/7068a0d09e61ec7424572b366751f048" target="_blank" rel="noopener noreferrer nofollow">https://gist.github.com/camperbot/7068a0d09e61ec7424572b366751f048</a>.
+Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/7068a0d09e61ec7424572b366751f048" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point </a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/serialization-of-a-user-object.md:31]:
```suggestion
Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/7068a0d09e61ec7424572b366751f048" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point</a>.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/set-up-a-template-engine.md [hunk 12]
```diff
@@ -28,7 +28,7 @@ Change the argument of the `res.render()` declaration in the `/` route to be the
 
 If all went as planned, your app home page will stop showing the message "`Pug template is not defined.`" and will now display a message indicating you've successfully rendered the Pug template!
 
-Submit your page when you think you've got it right. If you're running into errors, you can check out the project completed up to this point <a href="https://gist.github.com/camperbot/3515cd676ea4dfceab4e322f59a37791" target="_blank" rel="noopener noreferrer nofollow">https://gist.github.com/camperbot/3515cd676ea4dfceab4e322f59a37791</a>.
+Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/3515cd676ea4dfceab4e322f59a37791" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point </a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/set-up-a-template-engine.md:31]:
```suggestion
Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/3515cd676ea4dfceab4e322f59a37791" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point</a>.
```

File: curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/set-up-the-environment.md [hunk 13]
```diff
@@ -42,7 +42,7 @@ Now try loading up your app and authenticate and you should see in your server c
 
 **Note:**`io()` works only when connecting to a socket hosted on the same url/server. For connecting to an external socket hosted elsewhere, you would use `io.connect('URL');`.
 
-Submit your page when you think you've got it right. If you're running into errors, you can check out the project completed up to this point <a href="https://gist.github.com/camperbot/aae41cf59debc1a4755c9a00ee3859d1" target="_blank" rel="noopener noreferrer nofollow">https://gist.github.com/camperbot/aae41cf59debc1a4755c9a00ee3859d1</a>.
+Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/aae41cf59debc1a4755c9a00ee3859d1" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point </a>.
```
Comment (ieahleen) [inline curriculum/challenges/english/06-quality-assurance/advanced-node-and-express/set-up-the-environment.md:45]:
```suggestion
Submit your page when you think you've got it right. If you're running into errors, you can <a href="https://gist.github.com/camperbot/aae41cf59debc1a4755c9a00ee3859d1" target="_blank" rel="noopener noreferrer nofollow">check out the project completed up to this point</a>.
```

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]

**Review by majestic-owl448 — DISMISSED** [review]

**Comment by ieahleen** [comment]
Now, you need to resolve conflicts

**Review by ghost — COMMENTED** [review]
Looks great!!!

**Review by majestic-owl448 — APPROVED** [review]

**Review by ghost — COMMENTED** [review]
Solid Changes!

**Review by moT01 — APPROVED** [review]

**Pull request merged**